### PR TITLE
Added basic mermaid visualization to simulator.

### PR DESCRIPTION
**Pull request by mwhittaker**
There are many ways to visualize a history. First, we can use existing distributed tracing tools like Jaegar or Grafana. However, most of these tools look at a single trace at a time, whereas a history is a mix of many traces.

There is also [shiviz][], which is very close to what I want. Unfortunately, there is no notion of traces or spans, so it can be hard to follow when multiple requests interleave.

Finally, there are tools like [mermaid][]. These are very simple, but the resulting diagrams are hard to read and a bit janky. After some discussion, we decided to use mermaid for now, and invest in building our own visualization frontend later.

Here's an example visualization:

```mermaid
sequenceDiagram
    participant op1 as Op 1
    participant op2 as Op 2
    participant github.com/ServiceWeaver/weaver/internal/sim/div0 as sim.div 0
    participant github.com/ServiceWeaver/weaver/internal/sim/divMod0 as sim.divMod 0
    participant github.com/ServiceWeaver/weaver/internal/sim/identity0 as sim.identity 0
    participant github.com/ServiceWeaver/weaver/internal/sim/mod0 as sim.mod 0
    note right of op1: [1:1] divmod({11 71})
    op1->>github.com/ServiceWeaver/weaver/internal/sim/divMod0: [1:2] sim.divMod.DivMod(11, 71)
    note right of op2: [2:4] divmod({5 98})
    github.com/ServiceWeaver/weaver/internal/sim/divMod0->>github.com/ServiceWeaver/weaver/internal/sim/div0: [1:3] sim.div.Div(11, 71)
    github.com/ServiceWeaver/weaver/internal/sim/div0->>github.com/ServiceWeaver/weaver/internal/sim/identity0: [1:6] sim.identity.Identity(11)
    op2->>github.com/ServiceWeaver/weaver/internal/sim/divMod0: [2:5] sim.divMod.DivMod(5, 98)
    github.com/ServiceWeaver/weaver/internal/sim/identity0->>github.com/ServiceWeaver/weaver/internal/sim/div0: [1:6] return 11, <nil>
    github.com/ServiceWeaver/weaver/internal/sim/divMod0->>github.com/ServiceWeaver/weaver/internal/sim/div0: [2:7] sim.div.Div(5, 98)
    github.com/ServiceWeaver/weaver/internal/sim/div0->>github.com/ServiceWeaver/weaver/internal/sim/identity0: [1:8] sim.identity.Identity(71)
    github.com/ServiceWeaver/weaver/internal/sim/identity0->>github.com/ServiceWeaver/weaver/internal/sim/div0: [1:8] return 71, <nil>
    github.com/ServiceWeaver/weaver/internal/sim/div0->>github.com/ServiceWeaver/weaver/internal/sim/identity0: [2:9] sim.identity.Identity(5)
    github.com/ServiceWeaver/weaver/internal/sim/identity0->>github.com/ServiceWeaver/weaver/internal/sim/div0: [2:9] return 5, <nil>
    github.com/ServiceWeaver/weaver/internal/sim/div0->>github.com/ServiceWeaver/weaver/internal/sim/identity0: [2:10] sim.identity.Identity(98)
    github.com/ServiceWeaver/weaver/internal/sim/identity0->>github.com/ServiceWeaver/weaver/internal/sim/div0: [2:10] return 98, <nil>
    github.com/ServiceWeaver/weaver/internal/sim/div0->>github.com/ServiceWeaver/weaver/internal/sim/divMod0: [1:3] return 0, <nil>
    github.com/ServiceWeaver/weaver/internal/sim/div0->>github.com/ServiceWeaver/weaver/internal/sim/divMod0: [2:7] return 0, <nil>
    github.com/ServiceWeaver/weaver/internal/sim/divMod0->>github.com/ServiceWeaver/weaver/internal/sim/mod0: [1:11] sim.mod.Mod(11, 71)
    github.com/ServiceWeaver/weaver/internal/sim/mod0->>github.com/ServiceWeaver/weaver/internal/sim/identity0: [1:13] sim.identity.Identity(11)
    github.com/ServiceWeaver/weaver/internal/sim/divMod0->>github.com/ServiceWeaver/weaver/internal/sim/mod0: [2:12] sim.mod.Mod(5, 98)
    github.com/ServiceWeaver/weaver/internal/sim/identity0->>github.com/ServiceWeaver/weaver/internal/sim/mod0: [1:13] return 11, <nil>
    github.com/ServiceWeaver/weaver/internal/sim/mod0->>github.com/ServiceWeaver/weaver/internal/sim/identity0: [2:14] sim.identity.Identity(5)
    github.com/ServiceWeaver/weaver/internal/sim/identity0->>github.com/ServiceWeaver/weaver/internal/sim/mod0: [2:14] return 5, <nil>
    github.com/ServiceWeaver/weaver/internal/sim/mod0->>github.com/ServiceWeaver/weaver/internal/sim/identity0: [1:15] sim.identity.Identity(71)
    github.com/ServiceWeaver/weaver/internal/sim/mod0->>github.com/ServiceWeaver/weaver/internal/sim/identity0: [2:16] sim.identity.Identity(98)
    github.com/ServiceWeaver/weaver/internal/sim/identity0->>github.com/ServiceWeaver/weaver/internal/sim/mod0: [1:15] return 71, <nil>
    github.com/ServiceWeaver/weaver/internal/sim/mod0->>github.com/ServiceWeaver/weaver/internal/sim/divMod0: [1:11] return 11, <nil>
    github.com/ServiceWeaver/weaver/internal/sim/identity0->>github.com/ServiceWeaver/weaver/internal/sim/mod0: [2:16] return 98, <nil>
    github.com/ServiceWeaver/weaver/internal/sim/mod0->>github.com/ServiceWeaver/weaver/internal/sim/divMod0: [2:12] return 5, <nil>
    github.com/ServiceWeaver/weaver/internal/sim/divMod0->>op1: [1:2] return 0, 11, <nil>
    note right of op1: [1:1] return <nil>
    github.com/ServiceWeaver/weaver/internal/sim/divMod0->>op2: [2:5] return 0, 5, <nil>
    note right of op2: [2:4] return <nil>
```

[mermaid]: https://mermaid.js.org/
[shiviz]: https://bestchai.bitbucket.io/shiviz/

### CODE REVIEW

File: internal/sim/sim.go [hunk 1]
```diff
@@ -684,3 +688,85 @@ func returnError(component reflect.Type, method string, err error) []reflect.Val
...
+
+	// Create events.
+	for _, event := range r.History {
+		switch x := event.(type) {
+		case OpStart:
+			fmt.Fprintf(&b, "    note right of op%d: %s(%s)\n", x.TraceId, x.Name, commas(x.Args))
+		case OpFinish:
+			fmt.Fprintf(&b, "    note right of op%d: return %s\n", x.TraceId, x.Error)
+		case Call:
+			fmt.Fprintf(&b, "    note right of %s%d: block on %s.%s(%s)\n", x.Caller, x.Replica, shorten(x.Component), x.Method, commas(x.Args))
```
Comment (spetrovic77) [inline internal/sim/sim.go:755]: As discussed offline, these "block on" statements don't seem very helpful, as the user is only interested in the interleaving, and not where the framework blocked the operation.

**Review by spetrovic77 — APPROVED** [review]

File: internal/sim/sim.go (hunk 1 above)
Comment (mwhittaker) [reply internal/sim/sim.go]: Thanks Srdjan! The output is way less cluttered now.
